src/operators/switchAll.ts

```diff
@@ -58,13 +66,18 @@ class SwitchSubscriber<T> extends Subscriber<T> {
 }
 
 class InnerSwitchSubscriber<T> extends Subscriber<T> {
-  constructor(destination: Observer<any>, private parent: SwitchSubscriber<T>) {
-    super(destination);
```

Comment:
couldn't we still set `destination` on the `InnerSwitchSubscriber`, but invoke `super` with no argument? I believe the only logic we want to short-circuit here is the [shared `_subscription` assignment](https://github.com/ReactiveX/RxJS/blob/master/src/Subscriber.ts#L45).